components/backup_text/src/bin/rewrite/br_models.rs

```diff
@@ -93,40 +95,54 @@ impl BrColumnInfo {
 #[derive(Debug, Default, Clone)]
 pub struct RewriteInfo {
     pub table_info: TableInfo,
-    pub table_name: String,
-    pub db_name: String,
+    pub name: String,
 }
 
 impl From<kvproto::brpb::Schema> for RewriteInfo {
     fn from(schema: kvproto::brpb::Schema) -> Self {
-        let br_table_info = {
-            let str = String::from_utf8(schema.table).unwrap();
-            serde_json::from_str::<BrTableInfo>(&str)
+        let kvproto::brpb::Schema {
+            db,
+            table,
+            table_info,
+            ..
+        } = schema;
+
+        let db_name = {
+            let str = String::from_utf8(db).unwrap();
+            let info = serde_json::from_str::<BrDbInfo>(&str)
                 .map_err(|e| format!("{}\n{}", e, str))
-                .unwrap()
+                .unwrap();
+            info.name.original
         };
 
-        let br_db_info = {
-            let str = String::from_utf8(schema.db).unwrap();
-            serde_json::from_str::<BrDbInfo>(&str)
+        let br_table_info = {
+            let str = String::from_utf8(table).unwrap();
+            serde_json::from_str::<BrTableInfo>(&str)
                 .map_err(|e| format!("{}\n{}", e, str))
                 .unwrap()
         };
 
-        let table_name = br_table_info.name.original.clone();
-        let db_name = br_db_info.name.original;
-        let table_info = br_table_info.into_table_info_lossy();
+        let name = format!("{}.{}", db_name, br_table_info.name.original);
 
-        Self {
-            table_info,
-            table_name,
-            db_name,
-        }
+        let table_info = if table_info.is_empty() {
+            warn!(
+                "no tipb::TableInfo found, convert from br models";
+                "table" => &name,
+            );
+            br_table_info.into_table_info_lossy()
+        } else {
+            protobuf::parse_from_bytes::<TableInfo>(&table_info).unwrap()
```

Comment:
How about moving this line to the start of the function, so we can skip all other work `if !table_info.is_empty()`